kornia/filters/filter.py

```diff
@@ -175,7 +175,7 @@ def filter2d_separable(input: torch.Tensor,
                   [0., 5., 5., 5., 0.],
                   [0., 0., 0., 0., 0.]]]])
     """
-    out_x = filter2d(input, kernel_x.unsqueeze(0), border_type, normalized, padding)
+    out_x = filter2d(input, kernel_x.unsqueeze(1), border_type, normalized, padding)
```

Comment:
why is this needed to be changed ? can we provide a test ?

Reply:
It is needed because batch size should be the first dimension but before the function, despite what the documentation specified, expected the batch size to be 1

Reply:
If you look at the previous code, the function was called with `kernel_size[None]` meaning that the batch size was not considered which is no longer the case with this update.